src/index.ts

```diff
@@ -107,6 +113,23 @@ function swc(options: PluginOptions = {}): RollupPlugin {
           ? {}
           : getOptions(this, dirname(id), options.tsconfig);
 
+      let enaleExperimentalDecorators = false;
+      if (isTypeScript) {
+        try {
+          const tsPath = require.resolve('typescript', { paths: [process.cwd()] });
+          const packageJsonPath = lookup(tsPath, 'package.json');
+          const { version } = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
```

Comment:
If you are using `typescript` anyway, why not `require('typescript/package.json')`?